wme-sitebuilder/Wizards/GoLive.php

```diff
@@ -128,6 +179,194 @@ public function verifyDomain() {
+	 */
+	public function searchDomains() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return wp_send_json_error( new WP_Error(
+				'mapps-capabilities-failure',
+				__( 'You do not have permission to perform this action. Please contact a site administrator or log into the Nexcess portal to change the site domain.', 'nexcess-mapps' )
+			), 403 );
+		}
+
+		// Verify the domain structure.
+		$domain = ! empty( $_POST['domain'] )
+			? Domain::parseDomain( $_POST['domain'] )
+			: null;
+
+		$domain = $this->domains->formatDomain( $domain );
+
+		if ( empty( $domain ) ) {
+			return wp_send_json_error( new WP_Error(
+				'mapps-invalid-domain',
+				sprintf(
+					/* Translators: %1$s is the provided domain name. */
+					__( '"%s" is not a valid domain name. Please check your spelling and try again.', 'nexcess-mapps' ),
+					sanitize_text_field( $_POST['domain'] )
+				)
+			), 422 );
+		}
+
+		$response = $this->domains->searchAvailableDomains( $domain );
+
+		if ( is_wp_error( $response ) ) {
+			$companyName = _x( 'Nexcess', 'company name', 'nexcess-mapps' );
+			$companyName = apply_filters( 'nexcess_mapps_branding_company_name', $companyName );
+
+			return wp_send_json_error( new WP_Error(
+				'mapps-search-domains-failure',
+				sprintf(
+					/* Translators: %1$s is the branded company name, %2$s is the API error message. */
+					__( 'The %1$s API returned an error: %2$s', 'nexcess-mapps' ),
+					$companyName,
+					$response->get_error_message()
+				)
+			) );
+		}
+
+		$response_body = wp_remote_retrieve_body( $response );
+
+		return wp_send_json_success( $response_body );
+	}
+
+	/**
+	 * AJAX: create UI flow.
+	 */
+	public function createPurchaseFlow() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return wp_send_json_error( new WP_Error(
+				'mapps-capabilities-failure',
+				__( 'You do not have permission to perform this action. Please contact a site administrator or log into the Nexcess portal to change the site domain.', 'nexcess-mapps' )
+			), 403 );
+		}
+
+		$selected_domains = $_POST['domains'];
+		$request_domains  = [];
+
+		foreach ( $selected_domains as $selected_domain ) {
+			$domain_name = ! empty( $selected_domain['domain_name'] )
+				? Domain::parseDomain( $selected_domain['domain_name'] )
+				: null;
+
+			if ( empty( $domain_name ) || empty( $selected_domain['package_id'] ) ) {
+				continue;
+			}
+
+			$request_domains[] = [
+				'domain_name' => $domain_name,
+				'package_id'  => $selected_domain['package_id'],
+			];
+		}
+
+		if ( empty( $request_domains ) ) {
+			return wp_send_json_error( new WP_Error(
+				'mapps-invalid-domains',
+				sprintf(
+					__( 'No valid domain names provided.', 'nexcess-mapps' )
+				)
+			), 422 );
+		}
+
+		$this->getData()
+			->set( 'requested_domains', $request_domains )
+			->delete( 'purchased_domains' )
+			->save();
+
+		$return_url   = admin_url( 'admin.php?page=sitebuilder#/wizard/go-live?nexcess=true&step=3' );
```

Comment:
What does the `nexcess=true` mean in this context?

Reply:
@chumpgrub @jorgemasta Can you please explain this to Brian?

Reply:
Renaming `nexcess` parameter to `purchase`.